phoenix-core/src/test/java/org/apache/phoenix/replication/reader/ReplicationLogProcessorTest.java

```diff
@@ -0,0 +1,720 @@
+        replicationLogProcessor.closeReader(null);
+        Path filePath = new Path(testFolder.newFile("testCloseReader").toURI());
+        String tableName = "T_" + generateUniqueName();
+
+        // Create a valid log file with proper structure and one record
+        LogFileWriter writer = initLogFileWriter(filePath);
+
+        // Add a mutation to make it a proper log file with data
+        Mutation put = LogFileTestUtil.newPut("testRow", 1, 1);
+        writer.append(tableName, 1, put);
+        writer.sync();
+        writer.close();
+
+        // Test with valid reader
+        LogFileReader reader = Mockito.spy(new LogFileReader());
+
+        reader.init(new LogFileReaderContext(conf)
+                .setFileSystem(localFs)
+                .setFilePath(filePath));
+
+        replicationLogProcessor.closeReader(reader);
+
+        // Ensure reader's close method is called only once
+        Mockito.verify(reader, Mockito.times(1)).close();
+    }
+
+    /**
+     * Tests processing an empty mutation map - should complete without errors.
+     */
+    @Test
+    public void testProcessReplicationLogBatchWithEmptyMap() {
+        Map<String, List<Mutation>> emptyMap = new HashMap<>();
+
+        ReplicationLogProcessor replicationLogProcessor = new ReplicationLogProcessor(conf, executorService);
+        // Process empty batch - should not throw any exceptions and should return immediately
+        try {
+            replicationLogProcessor.processReplicationLogBatch(emptyMap);
+            // If we reach here, the empty map was processed successfully
+            assertTrue("Processing empty map should complete without errors", true);
+        } catch (Exception e) {
+            fail("Processing empty map should not throw exception: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Tests exception handling when attempting to process mutations for non-existent tables.
+     */
+    @Test
+    public void testProcessReplicationLogBatchExceptionsMessageIsCorrect() {
+        Map<String, List<Mutation>> tableMutationsMap = new HashMap<>();
+        Mutation mutation = LogFileTestUtil.newPut("abc", 6L, 5);
+        tableMutationsMap.put("NON_EXISTENT_TABLE", Collections.singletonList(mutation));
+        ReplicationLogProcessor replicationLogProcessor = new ReplicationLogProcessor(conf, executorService);
+        try {
+            replicationLogProcessor.processReplicationLogBatch(tableMutationsMap);
+            fail("Should throw TableNotFoundException for non-existent table");
+        } catch (IOException exception) {
+            assertTrue("Error message should mention file does not exist and file path name",
+                    exception.getMessage().contains("TableNotFoundException"));
+        }
+    }
+
+    /**
+     * Tests behavior when HBase operations fail (simulated by disabling table).
+     */
+    @Test
+    public void testProcessReplicationLogBatchWithHBaseFailure() throws Exception {
```

Comment:
It would also be useful to have a test case where only part of a batch fails. This means making a table with multiple regions, building a set of mutations that would be applied to all regions, and taking one of those regions offline.

Reply:
Right now we'd have an equivalent result... a thrown IOException. However as the implementation evolves we might have a more granular response. Resubmission of only the mutations involved with failed futures, etc. Do you want to try to tackle this now, even? It's much more likely a regionserver is down than a table is disabled, so batches are likely to only partially fail.